api/v1beta1/vaultauth_types.go

```diff
@@ -320,14 +320,23 @@ func (a *VaultAuthConfigGCP) Validate() error {
 type VaultAuthGlobalRef struct {
 	// Name of the VaultAuthGlobal resource.
 	// +kubebuilder:validation:Pattern=`^([a-z0-9.-]{1,253})$`
-	Name string `json:"name"`
+	Name string `json:"name,omitempty"`
 	// Namespace of the VaultAuthGlobal resource. If not provided, the namespace of
 	// the referring VaultAuth resource is used.
 	// +kubebuilder:validation:Pattern=`^([a-z0-9.-]{1,253})$`
 	Namespace string `json:"namespace,omitempty"`
 	// MergeStrategy configures the merge strategy for HTTP headers and parameters
 	// that are included in all Vault authentication requests.
 	MergeStrategy *MergeStrategy `json:"mergeStrategy,omitempty"`
+	// Default when set to true will use the default VaultAuthGlobal resource as the
+	// default if Name is not set. The "allow-global-defaults" option must be set on
+	// the operator's -global-vault-auth-options flag
+	//
+	// The VaultAuthGlobal search order is:
+	// 1. The VaultAuthGlobal in Namespace.
+	// 2. The VaultAuthGlobal in the VaultAuth Namespace.
```

Comment:
Doh, good catch.

Fixed in 494acf5dd37b924db88e5c0cf3537f803c1f4411

The search is now constrained to `.spec.vaultAuthGlobalResourceRef.namespace` when set, otherwise it searches the VA's namespace, then the operator's namespace.

Reply:
Nice! This is working for me now.